scipy/stats/_hypotests.py

```diff
@@ -1433,3 +1435,242 @@ def cramervonmises_2samp(x, y, method='auto'):
+        # error over the sample size.
+        tukey_criterion = srd * self._stand_err
+        # the confidence levels are determined by the
+        # `mean_differences` +- `tukey_criterion`
+        upper_conf = self.statistic + tukey_criterion
+        lower_conf = self.statistic - tukey_criterion
+        return ConfidenceInterval(low=lower_conf, high=upper_conf)
+
+
+def _tukeykramer_iv(args, sig_level):
+    args = [np.asarray(arg) for arg in args]
+    for arg in args:
+        if arg.ndim != 1:
+            raise ValueError("Input samples must be one-dimensional")
+        if arg.size <= 1:
+            raise ValueError("Input sample size must be greater than one.")
+        if np.isinf(arg).any():
+            raise ValueError("Input samples must be finite.")
+    if not 0 < sig_level < 1:
+        raise ValueError("Significance level should be between 0 and 1.")
+    return args
+
+
+def tukeykramer(*args, sig_level=.05):
+    """Perform the multiple comparison Tukey-Kramer test for equality of means.
+
+    Under the assumption of a previously rejected null hypothesis that two or
+    more samples have the same population mean, the Tukey-Kramer test compares
+    the absolute mean difference between each input group to the Tukey
+    criterion for significance. The null hypothesis for each comparison is that
+    they have equal means. For inputs of differing sample sizes, the
+    Tukey-Kramer method is used, albeit with a higher confidence coefficient,
+    leading to a more conservative estimate [1]_.
+
+    Parameters
+    ----------
+    sample1, sample2, ... : array_like
+        The sample measurements for each group. There must be at least
+        two arguments.
+    sig_level : float, optional
+        Significance level for which to determine the appropriate studentized
+        range distribution value.
+        Default is .05.
+
+    Returns
+    -------
+    result : `~scipy.stats._result_classes.TukeyKramerTestResult` instance
+        The return value is an object with the following attributes:
+
+        sig_level : float
+            The desired significance level (copied from `tukeykramer` input).
+        statistic : float ndarray
+            The computed statistic of the test for each comparison. The
+            (i, j) index is the statistic for the ith - jth group comparison.
+        pvalue : float ndarray
+            The associated p-value from the studentized range distribution. The
+            (i, j) index is the p-value for the ith - jth group comparison.
+
+
+        The object has the following methods:
+
+        confidence_ci() :
+            Compute the confidence interval for the significance level
+            specified in the test.
+
+    Notes
+    -----
+    The use of this test relies on several assumptions.
+
+    1. There are equal variances between the samples.
+    2. Each sample is from a normally distributed population.
```

Comment:
> Each sample is from a normally distributed population.

If the test is like the t-test, it's more general than that. The assumption is probably that _sample means_ of independent samples from the underlying distributions are normally distributed. This is true if the underlying distribution is normal, but also tends to be the case for many distributions due to the central limit theorem.